amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/client/EmailClient.kt

```diff
@@ -0,0 +1,48 @@
+package com.ritense.valtimo.amsterdam.emailapi.client
+
+import mu.KotlinLogging
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.client.RestTemplate
+import java.net.URI
+
+
+private val logger = KotlinLogging.logger {}
+
+class EmailClient(
+    private val restTemplate: RestTemplate,
```

Comment:
[RestTemplate is deprecated. Please use WebClient](https://medium.com/naukri-engineering/resttemplate-is-deprecated-use-webclient-a1dd7ca6394d)

Reply:
For compatibilty reasons in our project we have to use RestTemplate for now. Users can alter and use WebClient if their situation seems to fit.